### PR TITLE
KVM v12.3.0 with helm3 changes

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/releases/pull/459

Including only helm 3, containerlinux changes.

### CODE REVIEW

**Comment by tomahawk28** [comment]
Tested on `geckon`.

File: kvm/v12.3.0/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Please use [Upgrading tenant clusters to Helm 3](https://intranet.giantswarm.io/docs/dev-and-releng/helm/helm3-tenant-cluster-upgrade/) as a guide on the upgrade process for the checks and monitoring steps.
+
+**Note for future 12.x.x releases:**
+
+Please persist this note and the above, until all customers are on AWS v12.2.x and above.
+
+## Change details
+
+### app-operator [v2.3.1](https://github.com/giantswarm/app-operator/blob/master/CHANGELOG.md#231---2020-09-22)
```
Comment (rossf7) [inline kvm/v12.3.0/README.md:25]:
```suggestion
### app-operator [v2.3.3](https://github.com/giantswarm/app-operator/blob/master/CHANGELOG.md#233---2020-10-15)
```

**Review by rossf7 — APPROVED** [review]
LGTM

Just one nit on the readme.

File: kvm/v12.3.0/README.md (hunk 1 above)
Comment (tomahawk28) [reply kvm/v12.3.0/README.md:25]: thanks for finding it!

File: kvm/v12.3.0/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Please contact your Solution Engineer before upgrading. The upgrade is automated. However, it includes a data migration from Helm 2 release configmaps to Helm 3 release secrets, there are some pre-upgrade checks and we recommend monitoring the upgrade to ensure safety.
+
+**Note for Solution Engineers:**
+
+Please use [Upgrading tenant clusters to Helm 3](https://intranet.giantswarm.io/docs/dev-and-releng/helm/helm3-tenant-cluster-upgrade/) as a guide on the upgrade process for the checks and monitoring steps.
+
+**Note for future 12.x.x releases:**
+
+Please persist this note and the above, until all customers are on AWS v12.2.x and above.
```
Comment (cokiengchiara) [inline kvm/v12.3.0/README.md:21]:
```suggestion
Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+ Please persist this note until all customers are on KVM 12.1.x and above.
```
@giantswarm/app-squad-nginx , can you check this note is appropriate? This is related to making nginx optional in kvm 12.1.0

File: kvm/v12.3.0/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+## Change details
+
+### app-operator [v2.3.3](https://github.com/giantswarm/app-operator/blob/master/CHANGELOG.md#233---2020-10-15)
+### chart-operator [v2.3.5](https://github.com/giantswarm/chart-operator/blob/master/CHANGELOG.md#233---2020-09-29)
+- Updated Helm to [v3.3.4](https://github.com/helm/helm/releases/tag/v3.3.4).
+
+### kvm-operator [v3.12.2](https://github.com/giantswarm/kvm-operator/blob/master/CHANGELOG.md#3122---2020-10-16)
+- Added monitoring labels into prometheus metrics
+
+### containerlinux [2512.5.0](https://www.flatcar-linux.org/releases/#release-2512.5.0)
```
Comment (cokiengchiara) [inline kvm/v12.3.0/README.md:32]:
```suggestion
### Container Linux [2512.5.0](https://www.flatcar-linux.org/releases/#release-2512.5.0)
```

File: kvm/v12.3.0/announcement.md [hunk 4]
```diff
@@ -0,0 +1 @@
+**Giant Swarm release v12.3.0 for KVM is available.** This release upgrades all Helm releases managed by Giant Swarm to use Helm 3. Further details can be found in the [release notes](https://github.com/giantswarm/releases/tree/master/KVM/v12.3.0).
```
Comment (cokiengchiara) [inline kvm/v12.3.0/announcement.md:1]:
```suggestion
**Giant Swarm release v12.3.0 for KVM is available.** This release upgrades all Helm releases managed by Giant Swarm to use Helm 3. It also upgrades Container Linux to address security vulnerabilities. Further details can be found in the [release notes](https://github.com/giantswarm/releases/tree/master/KVM/v12.3.0).
```

@tfussell WDYT of including this here? Customers like knowing when security stuff is fixed right.

File: kvm/v12.3.0/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+## :zap:  Giant Swarm Release v12.3.0 for KVM :zap:
+
+This release upgrades all Helm releases managed by Giant Swarm to use [Helm v3.3.4](https://github.com/helm/helm/releases/tag/v3.3.4).
+
+This lets us benefit from the improved security model and keep up to date with the community. We also remove the Tiller deployment from the giantswarm namespace, removing its gRPC endpoint, which reduces operational complexity.
+
+If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3. As Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/
+
```
Comment (cokiengchiara) [inline kvm/v12.3.0/README.md:8]:
```suggestion

The release also upgrades Container Linux to address security vulnerabilities.

```

File: kvm/v12.3.0/README.md (hunk 2 above)
Comment (sslavic) [reply kvm/v12.3.0/README.md:21]: Adding that note makes sense, but I'm not sure it's on the right place.

```
Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+
```
could be part of the "Note for SEs", and then leave "Note for future 12.x.x releases:" unchanged

File: kvm/v12.3.0/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+Below, you can find more details on components that were changed with this release.
+
+**Note before upgrade:**
+
+Please contact your Solution Engineer before upgrading. The upgrade is automated. However, it includes a data migration from Helm 2 release configmaps to Helm 3 release secrets, there are some pre-upgrade checks and we recommend monitoring the upgrade to ensure safety.
+
+**Note for Solution Engineers:**
+
+Please use [Upgrading tenant clusters to Helm 3](https://intranet.giantswarm.io/docs/dev-and-releng/helm/helm3-tenant-cluster-upgrade/) as a guide on the upgrade process for the checks and monitoring steps.
```
Comment (cokiengchiara) [inline kvm/v12.3.0/README.md:17]:
```suggestion
Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+

Please use [Upgrading tenant clusters to Helm 3](https://intranet.giantswarm.io/docs/dev-and-releng/helm/helm3-tenant-cluster-upgrade/) as a guide on the upgrade process for the checks and monitoring steps.
```

File: kvm/v12.3.0/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+Below, you can find more details on components that were changed with this release.
+
+**Note before upgrade:**
+
+Please contact your Solution Engineer before upgrading. The upgrade is automated. However, it includes a data migration from Helm 2 release configmaps to Helm 3 release secrets, there are some pre-upgrade checks and we recommend monitoring the upgrade to ensure safety.
+
+**Note for Solution Engineers:**
+
+Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+
```
Comment (cokiengchiara) [inline kvm/v12.3.0/README.md:17]:
```suggestion
Before upgrading, please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first.
```

File: kvm/v12.3.0/README.md (hunk 2 above)
Comment (cokiengchiara) [reply kvm/v12.3.0/README.md:23]: @sslavic thanks that makes better sense. But should change "AWS v12.2.x" to "KVM v12.1.x+" right

Comment (sslavic) [reply kvm/v12.3.0/README.md:25]: Yes, AWS reference is wrong, it's KVM release

Comment (tomahawk28) [reply kvm/v12.3.0/README.md:25]: Can I commit what @cokiengchiara suggest earlier in this thread?

Comment (cokiengchiara) [reply kvm/v12.3.0/README.md:25]: thanks, I was asking to confirm if it should be 12.1, not 12.2 :D but I think that's right, so will commit this.

File: kvm/v12.3.0/README.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
+## :zap:  Giant Swarm Release v12.3.0 for KVM :zap:
+
+This release upgrades all Helm releases managed by Giant Swarm to use [Helm v3.3.4](https://github.com/helm/helm/releases/tag/v3.3.4).
+
+This lets us benefit from the improved security model and keep up to date with the community. We also remove the Tiller deployment from the giantswarm namespace, removing its gRPC endpoint, which reduces operational complexity.
+
+If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3. As Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/
```
Comment (tfussell) [inline kvm/v12.3.0/README.md:7]:
```suggestion
If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3 as Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/
```

**Review by cokiengchiara — APPROVED** [review]
LGTM

File: kvm/v12.3.0/release.diff [hunk 9]
```diff
@@ -0,0 +1,51 @@
+apiVersion: release.giantswarm.io/v1alpha1                         apiVersion: release.giantswarm.io/v1alpha1
+kind: Release                                                      kind: Release
+metadata:                                                          metadata:
+  creationTimestamp: null                                            creationTimestamp: null
+  name: v12.2.0                                                 |    name: v12.3.0
+spec:                                                              spec:
+  apps:                                                              apps:
+  - name: cert-exporter                                         |      - name: cert-exporter
```
Comment (tfussell) [inline kvm/v12.3.0/release.diff:8]: Could you fix the indentation of `release.yaml` for v12.3.0 so we don't have a diff showing every line changed?

File: kvm/v12.3.0/announcement.md [hunk 10]
```diff
@@ -0,0 +1 @@
+**Giant Swarm release v12.3.0 for KVM is available.** This release upgrades all Helm releases managed by Giant Swarm to use Helm 3. It also upgrades Container Linux to address security vulnerabilities. Further details can be found in the [release notes](https://github.com/giantswarm/releases/tree/master/KVM/v12.3.0).
```
Comment (tfussell) [inline kvm/v12.3.0/announcement.md:1]:
```suggestion
**Giant Swarm release v12.3.0 for KVM is available.** This release upgrades all Helm releases managed by Giant Swarm to use Helm 3. It also upgrades Container Linux to address security vulnerabilities. Further details can be found in the [release notes](https://github.com/giantswarm/releases/tree/master/kvm/v12.3.0).
```
This is case sensitive.

File: kvm/v12.3.0/release.diff (hunk 9 above)
Comment (tomahawk28) [reply kvm/v12.3.0/release.diff:8]: Good catch, i'm on it

**Review by tfussell — APPROVED** [review]
LGTM.

**Pull request merged**